src/services/poolServices/index.ts

```diff
@@ -461,3 +466,81 @@ export const checkWithdrawPoolLiquidityGasFee = async (
     payload: partialFee.toHuman(),
   });
 };
+
+export const createPoolCardsArray = async (
+  api: ApiPromise,
+  dispatch: Dispatch<PoolAction>,
+  pools: any,
+  selectedAccount?: InjectedAccountWithMeta
+) => {
+  const apiPool = api as ApiPromise;
+
+  try {
+    const poolCardsArray: PoolCardProps[] = [];
+
+    await Promise.all(
+      pools.map(async (pool: any) => {
+        const lpTokenId = pool?.[1]?.lpToken;
+
+        let lpToken = null;
+        if (selectedAccount?.address) {
+          const lpTokenAsset = await apiPool.query.poolAssets.account(lpTokenId, selectedAccount?.address);
+          lpToken = lpTokenAsset.toHuman() as LpTokenAsset;
+        }
+
+        if (pool?.[0]?.[1]?.interior?.X2) {
+          const poolReserve: any = await getPoolReserves(
+            apiPool,
+            pool?.[0]?.[1]?.interior?.X2?.[1]?.GeneralIndex?.replace(/[, ]/g, "")
```

Comment:
Can you make some helper function for replacing? You have replace(/[, ]/g, "") on many places.